### ISSUE TITLE
Bug (Neovim): Diagnostics not working

### ISSUE THREAD

**Issue by m0tholith** [nolanderc/glsl_analyzer#63]
Diagnostics don't really work on Neovim. Completion, documentation and everything else works well, attached is my LSP logfile:
[lsp.log](https://github.com/user-attachments/files/16228679/lsp.log)

**Comment by nolanderc** [comment]
Yeah, I never got around to publishing diagnostics to the client, even though we collect them during parsing.

Diagnostics are published as a LSP notification from the server to the client, and so far this isn't implemented, and thus would need some work.

With that implemented, we could send notifications upon receiving a `textDocument/didChange` or `textDocument/didOpen` request from the client. But we probably need some mechanism for queuing notifications, so that we don't send the notification until *after* having responded to the initial request.

I probably won't have the time to work on this for a while, but PRs are always welcome.
